### PR TITLE
Raised version to 3.3.1 and adapted old pyarrow patch

### PR DESCRIPTION
**Pull request by timostrunk**
The package offsets were broken. I fixed it and raised version to 3.3.1

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #141, #149 , #148,

### CODE REVIEW

**Comment by timostrunk** [comment]
I rewrote the arrow-cpp dependency to libarrow

**Comment by timostrunk** [comment]
@conda-forge-admin, please rerender

File: recipe/0001-drop-pyarrow-versions-build-with-proper-ABI-settings.patch [hunk 1]
```diff
@@ -1,18 +1,8 @@
-From 88d281e968026fd817c3bf65bbe0d3321cdbdff6 Mon Sep 17 00:00:00 2001
-From: Benjamin Zaitlen <quasiben@gmail.com>
-Date: Mon, 20 Mar 2023 08:36:54 +0100
```
Comment (xhochy) [inline recipe/0001-drop-pyarrow-versions-build-with-proper-ABI-settings.patch:3]: Please don't drop attribution and git structure on these patches.

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by timostrunk** [comment]
@conda-forge-admin, please rerender

**Comment by timostrunk** [comment]
I recreated this PR in #151 and removed unrequired patches. I would favor merging #151 and closing this one.

**Pull request closed without merging**